### DISCUSSION TITLE
How to improve the mounting speed of file system?

### DISCUSSION THREAD

**Discussion by 1414529969** [openzfs/zfs#13677] [Q&A]
As far as I know, ZFS file system will synchronize ZIL to disk before mounting. If there is too much data in ZIL, the ZFS file system will be mounted slowly. So, how to improve the mounting speed of file system?

**Comment by amotin** [comment]
ZIL has to be replayed during mount, unless you mount read-only.  You can reduce the amount of data in ZIL by clean pool export or reducing transaction group size by reducing amount of dirty data in ARC (in case you have too much RAM and slow storage).  But it is usually not so big problem, since ZIL covers only one TXG, which usually covers only few seconds of traffic.  There was scenario when repeating rewrite of the same data could stretch ZIL to huge sizes, but it was fixed some time ago.
